File: face-matcher/services/job_manager_service.py
import os

from loom24_shared.types import Job
from loom24_shared.services import create_service_client

_client = create_service_client(os.environ["JOB_MANAGER_URL"])


def get_job(job: Job) -> Job:
    data = _client.get(f"/get/id/{job.id}", job.userId)
    return Job.model_validate(data)

File: head-direction-checker/src/services/jobManagerService.ts
import { createServiceClient } from '@loom24/shared/services';
import { Job } from '@loom24/shared/types';

const client = createServiceClient(process.env.JOB_MANAGER_URL);

export const getJob = (job: Job): Promise<Job> =>
  client.get<Job>(`/get/id/${job.id}`, job.userId);
